scripts/install-plugin.sh

```diff
@@ -16,6 +16,59 @@ then
     exit 1
 fi
 
+function check_empty {
+    variable="$1"
+    message="$2"
+    if [ -z "$variable" ]
+    then
+        echo $message
+        exit 1
+    fi
+}
+
+[ -z "$terraform" ] && terraform=terraform
+# Getting the version without the initial "v"
+bare_version=$(echo $version | sed -e 's/^v//')
+
+# The default target directory is the pre-0.13 one
+target_dir=$HOME/.terraform.d/plugins/
+
+terraform_exec=""
+for dir in $(echo $PATH | tr ':' ' ')
+do
+    if [ -x $dir/$terraform ]
+    then
+        terraform_exec=$dir/$terraform
+        break
+    fi
+done
+
+if [ -z "$terraform_exec" ]
+then
+    echo "$terraform executable not found"
+    exit 1
+fi
+
+# Terraform version is used to determine what the target directory should be
+terraform_version=$($terraform version | head -n 1| sed -e 's/Terraform v//')
+check_empty "$terraform_version" "terraform_version not detected"
+
+
+terraform_major=$(echo $terraform_version | tr '.' ' '| awk '{print $1}')
+check_empty "$terraform_major" "terraform_version major not detected"
+terraform_minor=$(echo $terraform_version | tr '.' ' '| awk '{print $2}')
+check_empty "$terraform_minor" "terraform_version minor not detected"
+os=$(uname -s | tr '[A-Z]' '[a-z]')
+check_empty "$os" "operating system not detected"
+arch=${os}_amd64
+
+
+# if terraform executable is 0.13+, we use the new path
+if [[ $terraform_major -gt 0 || $terraform_major -eq 0 && $terraform_minor > 12 ]]
+then
+    target_dir=$HOME/.terraform.d/plugins/registry.terraform.io/vmware/vcd/$bare_version/$arch
```

Comment:
I'm trying this workflow:
1. Old (0.12) simple configuration with some catalog and catalog item.
2. Replace old terraform 0.12 binary with terraform 0.13 binary in the $PATH
3. Run `terraform init` and upgrade the config from 0.12 to 0.13
```
$ terraform init

Initializing the backend...

Initializing provider plugins...
- Using previously-installed -/vcd v2.9.0
- Finding latest version of hashicorp/vcd...

Error: Failed to install providers

Could not find required providers, but found possible alternatives:

  hashicorp/vcd -> terraform-providers/vcd

If these suggestions look correct, upgrade your configuration with the
following command:
    terraform 0.13upgrade .

$ terraform 0.13upgrade .

This command will update the configuration files in the given directory to use
the new provider source features from Terraform v0.13. It will also highlight
any providers for which the source cannot be detected, and advise how to
proceed.

We recommend using this command in a clean version control work tree, so that
you can easily see the proposed changes as a diff against the latest commit.
If you have uncommited changes already present, we recommend aborting this
command and dealing with them before running this command again.

Would you like to upgrade the module in the current directory?
  Only 'yes' will be accepted to confirm.

  Enter a value: yes

-----------------------------------------------------------------------------

Upgrade complete!

Use your version control system to review the proposed changes, make any
necessary adjustments, and then commit.
```
4. Run `make build` and `make install` from this PR:
```
$ make install
==> Checking that code complies with gofmt requirements...
go install -ldflags="-X 'github.com/vmware/terraform-provider-vcd/v3/vcd.BuildVersion=v1.0.0-438-gbbd4922'"
/Users/l/go/bin/terraform-provider-vcd -> /Users/l/.terraform.d/plugins/registry.terraform.io/vmware/vcd/3.0.0/darwin_amd64/terraform-provider-vcd_v3.0.0
Installed
total 66288
-rwxr-xr-x  1 l  33935556 Spa  7 18:19 terraform-provider-vcd_v3.0.0
```
5. Run `terraform init` with the hope that 3.0.0 will be used:
```
$ terraform init

Initializing the backend...

Initializing provider plugins...
- Using previously-installed terraform-providers/vcd v2.9.0
- Using previously-installed -/vcd v2.9.0

The following providers do not have any version constraints in configuration,
so the latest version was installed.

To prevent automatic upgrades to new major versions that may contain breaking
changes, we recommend adding version constraints in a required_providers block
in your configuration, with the constraint strings suggested below.

* -/vcd: version = "~> 2.9.0"
* terraform-providers/vcd: version = "~> 2.9.0"

Terraform has been successfully initialized!
```

What am I doing wrong? Why is it using published 2.9.0 instead of the built 3.0.0 and how can I fix this?
```

Reply:
The problem is that you need to run `make install` with terraform 0.13 to get the files in the right path. When you run that command with 0.12, the files go into the old path, where 0.13 doesn't find them.

The new `make install` looks at the terraform version and puts the files in the appropriate path.

Moreover, your HCL script needs to have a `terraform` block like this:

```
terraform {
  required_providers {
    vcd = {
      source = "vmware/vcd"
    }
  }
  # optional
  # required_version = ">= 0.13"
}
```

The files generated by our test scripts all have such block. But if you want to run it with your own scripts, you need to add it manually.

Lastly, you may need to add to the provider the property:

```
  version = "~> 3.0"
```

It should not be needed, but it might, depending on which other versions are already cached in your `$PWD/.terraform` directory.

Reply:
As a guideline, here's where `make install` produces when you have terraform 0.13.

```
.terraform.d/
├── checkpoint_signature
└── plugins
    ├── registry.terraform.io
    │   └── vmware
    │       └── vcd
    │           ├── 2.9.0
    │           │   └── darwin_amd64
    │           │       └── terraform-provider-vcd_v2.9.0
    │           └── 3.0.0
    │               └── darwin_amd64
    │                   └── terraform-provider-vcd_v3.0.0
```

If you are using Linux, you'll get a directory named `linux_amd64` instead

Reply:
> The problem is that you need to run make install with terraform 0.13 to get the files in the right path. When you run that command with 0.12, the files go into the old path, where 0.13 doesn't find them.

Yes, I actually was doing that.

> Moreover, your HCL script needs to have a terraform block like this:

It was really interesting. As I had run the `terraform 0.13upgrade` command, it did generate the `versions.tf` file with this block, but it contained the **old** 2.9 repo address (**not** in vmware namespace).

I believe we need a section in the docs explaining how to upgrade correctly.